src/Query/Builder.php

```diff
@@ -304,6 +304,33 @@ abstract public function count();
 
     abstract public function get($columns = ['*']);
 
+    public function when($value, $callback, $default = null)
+    {
+        if ($value) {
+            return $callback($this, $value) ?: $this;
+        } elseif ($default) {
+            return $default($this, $value) ?: $this;
+        }
+
+        return $this;
+    }
+
+    public function tap($callback)
+    {
+        return $this->when(true, $callback);
+    }
+
+    public function unless($value, $callback, $default = null)
+    {
+        if (! $value) {
+            return $callback($this, $value) ?: $this;
+        } elseif ($default) {
```

Comment:
same here, early return?

Reply:
I don't think they can as that affects the sequence of logic. As I said its been lifted straight from the laravel codebase.

Reply:
how can it affect the logic if there's a `return`?

Reply:
I've updated it now - ran the tests and seems ok.